code/cycler/src/wattrex_battery_cycler/mid/mid_pwr/mid_pwr.py

```diff
@@ -0,0 +1,230 @@
+                                        limit_type= self.actual_inst.limit_type,
+                                        limit_ref = self.actual_inst.limit_ref)
+            elif self.actual_inst.mode is CyclerDataPwrModeE.WAIT:
+                self.pwr_dev.set_wait_mode(time_ref= self.actual_inst.ref)
+            elif self.actual_inst.mode is CyclerDataPwrModeE.DISABLE:
+                self.pwr_dev.disable()
+            else:
+                log.error("The mode is not valid")
+                raise ValueError("The mode is not valid")
+            self.__get_pwr_direction()
+
+    def update_local_data(self, new_gen_meas: CyclerDataGenMeasC,
+                           new_status: CyclerDataAllStatusC) -> None:
+        """Function to update the local data with the given data
+
+        Args:
+            new_gen_meas (CyclerDataGenMeasC): [description]
+        """
+        self.local_gen_meas = new_gen_meas
+        self.local_status = new_status
+
+    def set_new_experiment(self, instructions: List[CyclerDataInstructionC],
+                        bat_pwr_range: CyclerDataPwrRangeC) -> None:
+        """Function to set a new experiment, it will clear the previous one
+        adding the new instruction set and battery limits
+
+        Args:
+            instructions (List[CyclerDataInstructionC]): [description]
+            bat_pwr_range (CyclerDataPwrRangeC): [description]
+        """
+        self.all_instructions = instructions
+        self.actual_inst.instr_id = -1
+        self.pwr_limits = bat_pwr_range
+
+    def process_iteration(self) -> Tuple[CyclerDataExpStatusE, int]:
+        """Processes a single instruction .
+
+        Returns:
+            Tuple[CyclerDataExpStatusE, int]: [description]
+        """
+        status = CyclerDataExpStatusE.QUEUED
+        # Check if the security limits are correct
+        if self.__check_security_limits():
+            # Clear differentiation if the experiment is done with epc or without
+            if self.pwr_dev.device_type is CyclerDataDeviceTypeE.EPC:
+                # The epc device always start in Disable mode,
+                # no need to check if instruction is not loaded
+                # When the epc goes back to disable means the last instruction is done
+                # if self.actual_inst.instr_id < 0:
+                if self.local_status.pwr_mode.value is CyclerDataPwrModeE.DISABLE.value:
+                    # Check if there are more instructions to read
+                    if len(self.all_instructions) > 0:
+                        self.actual_inst = self.all_instructions.pop(0)
+                        log.warning(f"New instruction: {self.actual_inst.__dict__}")
+                        self.__apply_instruction()
+                        status = CyclerDataExpStatusE.RUNNING
+                    else:
+                        self.actual_inst.instr_id = -1
+                        status = CyclerDataExpStatusE.FINISHED
+                else:
```

Comment:
Remove else

Reply:
Not possible as the status is a local variable that will be return to manager